src/Illuminate/Validation/Concerns/FormatsMessages.php

```diff
@@ -307,6 +311,35 @@ protected function replaceAttributePlaceholder($message, $value)
         );
     }
 
+    /**
+     * Replace the :index placeholder in the given message.
+     *
+     * @param  string  $message
+     * @param  string  $attribute
+     * @return string
+     */
+    protected function replaceIndexPlaceholder($message, $attribute)
+    {
+        $index = explode('.', $attribute);
+
+        return str_ireplace(':index', $index[1] ?? 0, $message);
+    }
+
+    /**
+     * Replace the :pos placeholder in the given message.
+     *
+     * @param  string  $message
+     * @param  string  $attribute
+     * @return string
+     */
+    protected function replacePositionPlaceholder($message, $attribute)
+    {
+        $index = explode('.', $attribute);
+        $i = ($index[1] ?? 0) + 1;
```

Comment:
if the $attribute is string then exploding it should return array of strings then you try to increase a string so it will throw an error. You may need to cast it to int. And you don't need $i variable you can write it directly inside irreplace